paddleseg/models/espnetv1.py

```diff
@@ -0,0 +1,276 @@
+# limitations under the License.
+
+import paddle
+import paddle.nn as nn
+import paddle.nn.functional as F
+import os
+
+from paddleseg.models import layers
+from paddleseg.cvlibs import manager
+from paddleseg.utils import utils
+
+def SyncBatchNorm(*args, **kwargs):
+    """In cpu environment nn.SyncBatchNorm does not have kernel so use nn.BatchNorm2D instead"""
+    if paddle.get_device() == 'cpu' or os.environ.get('PADDLESEG_EXPORT_STAGE'):
+        return nn.BatchNorm2D(*args, **kwargs)
+    elif paddle.distributed.ParallelEnv().nranks == 1:
+        return nn.BatchNorm2D(*args, **kwargs)
+    else:
+        return nn.SyncBatchNorm(*args, **kwargs)
+
+ 
+@manager.MODELS.add_component
+class ESPNetV1(nn.Layer):
+    """
+    The ESPNetV1 implementation based on PaddlePaddle.
+    The original article refers to
+    Sachin Mehta1, Mohammad Rastegari, Anat Caspi, Linda Shapiro, and Hannaneh Hajishirzi. "ESPNet: Efficient Spatial Pyramid of Dilated Convolutions for Semantic Segmentation"
+    (https://arxiv.org/abs/1803.06815).
+    Args:
+        num_classes (int): The unique number of target classes.
+        in_channels (int, optional): Number of input channels. Default: 3.
+        level2_depth (int, optional): Depth of DilatedResidualBlock. Default: 2.
+        level3_depth (int, optional): Depth of DilatedResidualBlock. Default: 3.
+        pretrained (str, optional): The path or url of pretrained model. Default: None.
+    """
+    def __init__(self, num_classes, in_channels=3, level2_depth=2, level3_depth=3, pretrained=None):
+        super().__init__()
+        self.encoder = ESPNetEncoder(num_classes, in_channels, level2_depth, level3_depth)
+
+        self.level3_up = nn.Conv2DTranspose(num_classes, num_classes, 2, stride=2, padding=0, output_padding=0, bias_attr=False)
+        self.br3 = SyncBatchNorm(num_classes)
+        self.level2_proj = Conv(in_channels + 128, num_classes, 1, 1)
+        self.combine_l2_l3 = nn.Sequential(
+            BNPReLU(2 * num_classes),
+            DilatedResidualBlock(2 * num_classes, num_classes, residual=False),
+        )
+        self.level2_up = nn.Sequential(
+            nn.Conv2DTranspose(num_classes, num_classes, 2, stride=2, padding=0, output_padding=0, bias_attr=False),
+            BNPReLU(num_classes),
+        ) 
+        self.out_proj = ConvBNPReLU(16 + in_channels + num_classes, num_classes, 3, 1)
+        self.out_up = nn.Conv2DTranspose(num_classes, num_classes, 2, stride=2, padding=0, output_padding=0, bias_attr=False)
+    
+    def forward(self, x):
+        p1, p2, p3 = self.encoder(x) # shape [N, C, H/2, W/2]  [N, C, H/4, W/4]  [N, C, H/8, W/8] 
+        up_p3 = self.level3_up(p3) # [N, C, H/4, W/4]
+
+        combine = self.combine_l2_l3(paddle.concat([up_p3, p2], axis=1))
+        up_p2 = self.level2_up(combine)  # [N, C, H/2, W/2]
+
+        combine = self.out_proj(paddle.concat([up_p2, p1], axis=1))  # shape [N, C, H/2, W/2]
```

Comment:
Remove this annotation please, here and elsewhere

Reply:
fixed